File: java/oop/RectangleApp/src/com/techlabs/rectangle/Rectangle.java
package com.techlabs.rectangle;

public class Rectangle {
	//for testing reflection
	public int x,y;
	private int width=20;
	private int height=20;
	private String borderColor="Default color";
	
	//for testing reflection
	public Rectangle(){
		System.out.println("Object created");
	}
	//for testing reflection
	public Rectangle(int x,int y){
		System.out.println("Object created");
	}
	
	public int calculateArea(){
		return width*height;
	}
	
	public void setborderColor(String borderColor){
		this.borderColor =checkBorderColor(borderColor);
	
	}
	
	public void setWidth(int width){
			this.width=checkWidthHeight(width);	
	}
	public void setHeight(int height){
		this.height=checkWidthHeight(height);	
	}
	
	public int getWidth(){
		return this.width;
	}
	
	public int getHeight(){
		return this.height;
	}
	
	public String getBorderColor(){
		return this.borderColor;
	}
	
	private int checkWidthHeight(int value){
		if(value>100){
			return 100;
		}else if(value<1){
			return 1;
		}else{
			return value;
		}
	}
	
	private String checkBorderColor(String bordercolor){
		if(bordercolor.equalsIgnoreCase("red")){
			return "Red";
		}else if(bordercolor.equalsIgnoreCase("blue")){
			return "blue";
		}else if(bordercolor.equalsIgnoreCase("Green")){
			return "Green";
		}else{
			return "No colors";
		}
	}
	
	
	/*public void setHeight(int height){
		if(height>100){
			this.height=100;
		}else if(height<1){
			this.height=1;
		}else{
			this.height=height;
		}
		
	} */
}

File: dotNet/csharp/RectangleApp/RectangleApp/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RectangleApp
{
    public class Rectangle
    {
        private int _width;
        private int _height;
        private string _BorderColor = "Default Color";


        public Rectangle()
        {
            Console.WriteLine("Object created....");
        }

        public int Width
        {
            get
            {
                return this._width;
            }
            set
            {
                int val= CheckWidthHeight(value);
                this._width = val;
            }
        }
        public int Height
        {
            get
            {
                return this._height;
            }
            set
            {
                int val = CheckWidthHeight(value);
                this._height = val;
            }
        }

        public string BorderColor
        {
            get
            {
                return this._BorderColor;
            }
            set
            {
                string color = CheckBorderColor(value);
                this._BorderColor = color;
            }
        }

       
        public int CalculateArea()
        {
            return Width * Height;

        }

        private int CheckWidthHeight(int val)
        {
            if (val>100)
            {
                return 100;
            }else if (val < 1)
            {
                return 1;
            }else
            {
                return val;
            }
        }

        private string CheckBorderColor(string color)
        {
            if (color.Equals("Red",StringComparison.CurrentCultureIgnoreCase))
            {
                return "Red";
            }
            else if (color.Equals("blue", StringComparison.CurrentCultureIgnoreCase))
            {
                return "blue";
            }
            else if(color.Equals("Green", StringComparison.CurrentCultureIgnoreCase))
            {
                return "Green";
            }
            else
            {
                return "No colors";
            }
        }

    }
}
